elifetools/utils_html.py

```diff
@@ -106,6 +108,23 @@ def replace_xref_tags(s):
 
     return s
 
+def replace_related_object_tags(s):
+    for tag_match in re.finditer("<(related-object.*?)>", s):
+        xlink_match = re.finditer('xlink:href="(.*)"', tag_match.group())
+        if xlink_match:
+            try:
+                xlink = next(xlink_match).group(1)
+                if not xlink[0:4] in ['http', 'ftp:']:
+                    xlink = 'http://' + xlink
+                new_tag = '<a href="' + xlink + '">'
+                old_tag = '<' + tag_match.group(1) + '>'
+                s = s.replace(old_tag, new_tag)
+                # Replace all close tags even if one open tag gets replaced
+                s = replace_simple_tags(s, 'related-object', 'a')
+            except StopIteration:
```

Comment:
is `replace_related_object_tags` being called within a generator?

Reply:
I think in these cases, the result of `re.finditer()` is the generator, on which `next()` is called in the loop, at least I think that's the situation.

Reply:
I'm not completely sure how generators differ from iterators, but the docs say `re.finditer()` returns an interator. Do the generators define iterators? I don't think there is anything more complicated in other layers of how these function calls are done, it is just relying on the `next()` and `StopIteration` exception to loop through the matches.

Reply:
> it is just relying on the next() and StopIteration exception to loop through the matches.

cheers, was just curious